subsys/nrf_security/src/utils/nrf_security_mutexes.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <errno.h>
+#include <stdbool.h>
+#include <zephyr/kernel.h>
+#include <nrf_security_mutexes.h>
+
+#if defined(CONFIG_MULTITHREADING) && !defined(__NRF_TFM__)
+
+int nrf_security_mutex_lock(nrf_security_mutex_t mutex)
+{
+	return k_mutex_lock(mutex.zephyr_mutex, K_FOREVER);
+}
+
+int nrf_security_mutex_unlock(nrf_security_mutex_t mutex)
+{
+	return k_mutex_unlock(mutex.zephyr_mutex);
+}
+
+#else
+
+int nrf_security_mutex_lock(nrf_security_mutex_t mutex)
+{
+	(void) mutex;
+	return 0;
+}
+
+int nrf_security_mutex_unlock(nrf_security_mutex_t mutex)
+{
+	(void) mutex;
+	return 0;
+}
+#endif
```

Comment:
I dont see the need for a struct that contains a single item.

I think a pointer to a typedef would work fine and be simpler.

But perhaps there is some problem with this that I cant see.

Reply:
I converted everything to typedefs now